src/storage/src/flush.rs

```diff
@@ -236,8 +243,11 @@ impl<S: LogStore> FlushJob<S> {
 impl<S: LogStore> Job for FlushJob<S> {
     // TODO(yingwen): [flush] Support in-job parallelism (Flush memtables concurrently)
     async fn run(&mut self, ctx: &Context) -> Result<()> {
-        let file_metas = self.write_memtables_to_layer(ctx).await?;
-
+        let res = self.write_memtables_to_layer(ctx).await;
+        for cb in self.flush_callbacks.iter() {
+            cb(res.is_ok(), self.flush_sequence).await;
```

Comment:
We shouldn't call the callback (mark stable) before `write_manifest_and_apply()`. As we should delete the log entries after the flush is applied (persisting into the manifest).

I'd suggest invoking `mark_stable` at the `RegionWriter`, after `version_control.apply_edit(version_edit)`
https://github.com/GreptimeTeam/greptimedb/blob/9ad6ddb26e05b70471c11e8ad78cc87723897fd5/src/storage/src/region/writer.rs#L131